src/dbnode/storage/series/reader.go

```diff
@@ -208,7 +217,15 @@ func (r Reader) fetchBlocksWithBlocksMapAndBuffer(
 	nsCtx namespace.Context,
 ) ([]block.FetchBlockResult, error) {
 	var (
-		// TODO(r): pool these results arrays
+		// Two-dimensional slice such that the first dimension is unique by blockstart
+		// and the second dimension is blocks of data for that blockstart (not necessarily
+		// in chronological order).
+		//
+		// ex. (querying 2P.M -> 6P.M with a 2-hour blocksize):
+		// [][]block.FetchBlockResult{
```

Comment:
This is actually a 1D slice which is one block start per element.

Reply:
Refactored to be more clear, take another look